index.bs

```diff
@@ -52,10 +66,169 @@ dictionary StorageBucketOptions {
 </xmp>
 
 <h3 id="storage-bucket-open">Creating a bucket</h3>
+
+<div algorithm>
+
+The <dfn method for="StorageBucketManager">open(|name|, |options|)</dfn> method steps are:
+
+1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+
+1. Let |storageKey| be the result of running [=obtain a storage key=] given |environment|.
+
+1. If |storageKey| is failure, then [=exception/throw=] a "{{SecurityError}}" {{DOMException}} and abort these steps.
```

Comment:
As I understand it, a `TypeError` is for when there's an invalid parameter.  I believe this text was pretty much lifted from IDB, e.g. https://www.w3.org/TR/IndexedDB/#dom-idbfactory-deletedatabase

However I agree it is a bit odd because the steps to obtain a storage key don't seem to mention anything about a secure context, just opaque origins and I can't tell exactly what those are. Implementation wise, you can see that navigator.storageBuckets is restricted to SecureContexts and there's an opaque origin check as well which throws this security error (seems to be true across all storage APIs).

@ayuishii

Reply:
This specifically has been updated to retrieve |shelf| instead, and updated to return `UnknownError`... but following the guidance given here about error mappings [1], which mentions that it recommends specific errors depending on the scenario. Although this is probably not followed consistently in practice, curious about your thoughts and if there is more I can learn about the TypeError recommendation?

[1] https://github.com/whatwg/fs/issues/57#issuecomment-1277022240